master/docs/manual/customization.rst

```diff
@@ -932,18 +932,153 @@ For more information on the available commands, see :doc:`../developer/master-wo
+This allows it to be reactive and real time, but comes at a price of a fair complexity.
+Sometimes, you just need a dashboard displaying your build results in your own weird manner but learning AngularJS for that is just too much.
+
+There is a Buildbot plugin which allows to write a server side generated dashboard, and integrate it in the UI.
+
+.. code-block:: python
+
+    # This needs buildbot and buildbot_www >= 0.9.5
+    pip install buildbot_wsgi_dashboards flask
+
+- This plugin can use any WSGI compatible web framework, Flask_ is a very common one, Bottle_ is another popular option.
+
+- The application needs to implement a ``/index.html`` route, which will render the html code representing the dashboard.
+
+- The application framework runs in a thread outside of Twisted.
+  No need to worry about Twisted and asynchronous code.
+  You can use python-requests_ or any library from the python ecosystem to access other servers.
+
+- You could use Http in order to access Buildbot :ref:`REST_API`, but you can also use the :ref:`Data_API`, via the provided synchronous wrapper.
+
+    .. py:method:: buildbot_api.dataGet(path, filters=None, fields=None, order=None, limit=None, offset=None):
+
+        :param tuple path: A tuple of path elements representing the API path to fetch.
+            Numbers can be passed as strings or integers.
+        :param filters: result spec filters
+        :param fields: result spec fields
+        :param order: result spec order
+        :param limit: result spec limit
+        :param offset: result spec offset
+        :raises: :py:exc:`~buildbot.data.exceptions.InvalidPathError`
+        :returns: a resource or list, or None
+
+        This is a blocking wrapper to master.data.get as described in :ref:`Data_API`.
+        The available paths are described in the :ref:`REST_API`, as well as the nature of return values depending on the kind of data that is fetched.
+        Path can be either the REST path e.g. ``"builders/2/builds/4"`` or tuple e.g. ``("builders", 2, "builds", 4)``.
+        The latter form being more convenient if some path parts are coming from variables.
+        The :ref:`Data_API` and :ref:`REST_API` are functionally equivalent except:
+
+        - :ref:`Data_API` does not have HTTP connection overhead.
+        - :ref:`Data_API` does not enforce authorization rules.
+
+        ``buildbot_api.dataGet`` is accessible via the WSGI application object passed to ``wsgi_dashboards`` plugin (as per the example).
+
+- That html code output of the server runs inside AngularJS application.
+
+  - It will use the CSS of the AngularJS application (including the Bootstrap_ css base).
+  - It can use some of the AngularJS directives defined by buildbot UI (currently only buildsummary is usable).
+  - It has full access to the application JS context.
+
+Here is an example of code that you can use in your master.cg to create a simple dashboard:
+
+.. code-block:: python
```

Comment:
Has this full block been confirmed to work? Would it be possible to add a test for it?